Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -35,44 +37,52 @@ This result is not yet formalised.
 * `Module.Flat.directSum`: arbitrary direct sums of flat modules are flat
 * `Module.Flat.of_free`: free modules are flat
 * `Module.Flat.of_projective`: projective modules are flat
+* `Module.Flat.iff_rTensor_preserves_injective_linearMap`: a module is flat iff tensoring preserves
+  injectiveness.
+
+## Implementation notes
+In `Module.Flat.iff_rTensor_preserves_injective_linearMap`, we require that the universe level of
+the ring is lower than or equal to that of the module. This requirement is to make sure ideals of
+the ring can be lifited to the universe of the module. It is unclear if this lemma also holds
+when module lives in a lower universe.
+
+This requirement also appears in `Algebra/ModuleCat/Injective`.
+There are some ideas proposed by Junyan Xu about potentially circumvent at
+[here](https://github.com/leanprover-community/mathlib4/pull/8905#discussion_r1428509361)
 
 ## TODO
 
-* Show that tensoring with a flat module preserves injective morphisms.
-  Show that this is equivalent to be flat.
-  See <https://stacks.math.columbia.edu/tag/00HD>.
-  To do this, it is probably a good idea to think about a suitable
-  categorical induction principle that should be applied to the category of `R`-modules,
-  and that will take care of the administrative side of the proof.
 * Show that flatness is stable under base change (aka extension of scalars)
   For base change, it will be very useful to have a "characteristic predicate"
   instead of relying on the construction `A ⊗ B`.
   Indeed, such a predicate should allow us to treat both
```

Comment:
[IsBaseChange](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IsTensorProduct.html#IsBaseChange) is now in mathlib, so maybe
```suggestion
  Using the `IsBaseChange` predicate should allow us to treat both
```